backend/api/views.py

```diff
@@ -40,10 +45,35 @@ def getEventByID(request, id):
 @api_view(['DELETE'])
 def deleteEvent(request, id):
     try:
+        get_list_or_404(Event, id=id)
         Event.objects.filter(id=id).delete()
+        return HttpResponseRedirect("/")
     except ObjectDoesNotExist:
         print("Event does not exist")
 
+# gets the non registered users by event id
+@api_view(['GET'])
+def get_non_registered_users_by_event_id(request, id):
+    try:
+        event = Event.objects.get(id=id)
+        non_registered_users = event.nonregistereduser_set.all()
+        serializer = NonRegisteredUserSerializer(non_registered_users, many=True)
+        return Response(serializer.data)
+    except ObjectDoesNotExist:
+        print("non registered users or event does not exist")
+
+@api_view(['GET'])
+def get_non_registered_users(request):
+    non_registered_users = NonRegisteredUser.objects.all()
+    serializers = NonRegisteredUserSerializer(non_registered_users, many=True)
+    return Response(serializers.data)
 
-    
-    
\ No newline at end of file
+@api_view(['POST'])
+def add_non_registered_user(request):
+    serializer = NonRegisteredUserSerializer(data=request.data)
+    if serializer.is_valid():
+        serializer.save()
+        print("saved")
+    else:
+        print("invalid: ", serializer.errors)
```

Comment:
I think we can remove these because it was for testing purposes?

Reply:
I think its important to keep the error printing and make It a try catch block later on which prints out the correct HTTP error